<a href="https://colab.research.google.com/github/ignacionachorodriguez/entrega1-turismo/blob/main/entrega1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Entrega 1 - PAD 2 (2026)**

## Grupo 2

- Fabián Iglesias
- Cecilia Venturino
- Keylin Fuentes
- Juan Ignacio Rodriguez


---

## Dataset

- **Nombre del dataset:** emisivo.xlsx
- **Tipo de archivo:** Excel (.xlsx)
- **Fuente:** Catálogo de Datos Abiertos del Gobierno de Uruguay.

### Descripción

Este conjunto de datos contiene información sobre el **Turismo Emisivo** de Uruguay. Incluye información de los residentes uruguayos que viajan al exterior y el gasto realizado según el destino principal del viaje.

Los datos se encuentran disponibles desde el año 2016 e incluyen variables como:

- Transporte internacional de salida e ingreso.
- Fechas de salida y entrada.
- Nacionalidad.
- Departamento de residencia.
- Motivo del viaje.
- Ocupación y nivel educativo del viajero.
- Destino principal.
- Tipo de alojamiento.
- Transporte local.
- Estadía.
- Gastos desagregados en alojamiento, alimentación, transporte, transporte local, cultura, tours, compras, entre otros.

Este conjunto de datos permite analizar el flujo de viajes de los residentes uruguayos al exterior, conocer las principales características de esos viajes y estimar el gasto realizado durante los mismos.

---

## Repositorio

Con el objetivo de facilitar la reproducibilidad del análisis y el trabajo colaborativo, el notebook y el dataset fueron alojados en un repositorio público de GitHub. El archivo "emisivo.xlsx" se carga directamente desde dicho repositorio mediante una URL pública, permitiendo ejecutar el notebook en Google Colab sin necesidad de descargar archivos manualmente.

##1. Cargar y visualizar el dataset

Carga tu dataset en un DataFrame y explora cómo luce visualizando algunas filas (puedes elegir mostrar las primeras y/o las últimas).

In [22]:
import pandas as pd # Importamos la biblioteca pandas

url = "https://raw.githubusercontent.com/ignacionachorodriguez/entrega1-turismo/main/datos/emisivo.xlsx" # URL del archivo EXCEL alojado en el repositorio de Github

df_emisivo = pd.DataFrame(pd.read_excel(url)) # Leemos el archivo EXCEL y lo cargamos en un DataFrame

df_emisivo.head() # Mostramos las primeras 5 filas del DataFrame

,IdLugarSalida,Lugar Salida,IdTranspSalidad,Transporte Internacional de Salida,FechaSalida,IdFecSalida,FechaEntrada,IdFecEntrada,IdNacionalidad,Pais,...,GastoAlojamiento,GastoAlimentacion,GastoTransporteInternac,GatoTransporteLocal,GastoCultural,GastoTours,GastoCompras,GastoResto,Coef,CoefTot
0,4,Chuy,2,Terrestre Auto,2016-12-29,13512,2017-01-10,13524,1,Uruguay,...,981.39,350.50,0.0,175.24,0.00,0.00,0.00,0.00,89.61,448.05
1,18,Río Branco,2,Terrestre Auto,2016-12-29,13512,2017-01-09,13523,1,Uruguay,...,1100.00,880.00,0.0,385.00,300.00,235.00,210.00,165.00,96.18,480.88
2,3,Aeropuerto de Carrasco,1,Aereo,2017-01-03,13517,2017-01-27,13541,1,Uruguay,...,0.00,1129.75,214.0,836.85,523.03,256.29,2039.83,0.00,129.04,258.09
3,3,Aeropuerto de Carrasco,1,Aereo,2017-01-06,13520,2017-01-21,13535,1,Uruguay,...,2228.41,1782.73,117.0,779.94,445.68,0.00,311.98,334.26,129.04,387.13
4,4,Chuy,2,Terrestre Auto,2017-01-02,13516,2017-01-25,13539,1,Uruguay,...,1800.00,933.12,0.0,408.24,152.14,0.00,106.50,0.00,89.61,448.05


In [23]:
df_metadatos = pd.read_excel(url, sheet_name="Metadatos") # También leemos la segunda página del archivo EXCEL que contiene los Metadatos, y lo cargamos en otro DataFrame

df_metadatos

,nombreDeAtributo,descripcion,tipoDeDato,recursoRelacionado,informacionAdicional
0,IdLugarSalida,Es el código del lugar de salida,Numeric,NaN,NaN
1,Lugar Salida,Es el lugar de salida,String,NaN,NaN
2,IdTranspSalidad,Es el código de transporte para el egreso,Numeric,NaN,NaN
3,Transporte Internacional de Salida,Transporte Internacional de egreso,String,NaN,NaN
4,FechaSalida,Fecha de salida,Date,NaN,NaN
5,IdFecSalida,Es el código de la fecha de salida,Numeric,NaN,NaN
6,FechaEntrada,Fecha de entrada,Date,NaN,NaN
7,IdFecEntrada,Es el código de la fecha de entrada,Numeric,NaN,NaN
8,IdNacionalidad,Es el código de la nacionalidad de quien egresó,Numeric,NaN,NaN
9,Pais,Es el país a donde se viajo,String,NaN,NaN


## 2. Dimensiones del dataset

Analiza cuántas filas y columnas tiene tu dataset.

In [24]:
print(df_emisivo.shape) # .shape devuelve una tupla con el número de filas y columnas del DataFrame

num_filas, num_columnas = df_emisivo.shape # Asignamos cada valor a una variable

print(f"El dataset tiene {num_filas} filas y {num_columnas} columnas.") # Mostramos el resultado utilizando una f-string

(32109, 41)
El dataset tiene 32109 filas y 41 columnas.


## 3. Información general del dataset

Obtén un resumen sobre el tipo de datos de cada columna, cuántos valores no nulos tiene y otra información relevante.

In [25]:
df_emisivo.info() # Obtenemos un resumen general del DataFrame.

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32109 entries, 0 to 32108
Data columns (total 41 columns):
 #   Column                               Non-Null Count  Dtype         
---  ------                               --------------  -----         
 0   IdLugarSalida                        32109 non-null  int64         
 1   Lugar Salida                         32109 non-null  object        
 2   IdTranspSalidad                      32109 non-null  int64         
 3   Transporte Internacional de Salida   32109 non-null  object        
 4   FechaSalida                          32109 non-null  datetime64[ns]
 5   IdFecSalida                          32109 non-null  int64         
 6   FechaEntrada                         32109 non-null  datetime64[ns]
 7   IdFecEntrada                         32109 non-null  int64         
 8   IdNacionalidad                       32109 non-null  int64         
 9   Pais                                 32109 non-null  object        
 10  IdDeptoRes

##4. Estadísticas descriptivas

Obtén un resumen estadístico de las columnas numéricas y, si es posible, de las categóricas.

In [26]:
df_emisivo.describe() # Resumen estadístico de las columnas numéricas

,IdLugarSalida,IdTranspSalidad,FechaSalida,IdFecSalida,FechaEntrada,IdFecEntrada,IdNacionalidad,IdDeptoResidencia,IdMotivo,IdOcupacion,...,GastoAlojamiento,GastoAlimentacion,GastoTransporteInternac,GatoTransporteLocal,GastoCultural,GastoTours,GastoCompras,GastoResto,Coef,CoefTot
count,32109.000000,32109.000000,32109,32109.000000,32109,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000,...,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000,32109.000000
mean,6.095799,2.603881,2021-02-23 18:04:59.355321088,15029.753465,2021-03-02 04:20:57.965056512,15036.181226,1.469962,28.495655,2.112648,14.805942,...,322.235384,278.991763,319.589442,138.286603,84.889594,5.363120,254.353178,112.580911,243.475173,553.616123
min,1.000000,1.000000,2016-12-13 00:00:00,13496.000000,2017-01-02 00:00:00,13516.000000,1.000000,0.000000,1.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.550000,1.550000
25%,3.000000,2.000000,2018-03-27 00:00:00,13965.000000,2018-04-04 00:00:00,13973.000000,1.000000,20.000000,1.000000,11.000000,...,0.000000,94.320000,0.000000,18.770000,0.000000,0.000000,33.030000,0.000000,119.940000,214.210000
50%,7.000000,2.000000,2020-02-13 00:00:00,14653.000000,2020-02-19 00:00:00,14659.000000,1.000000,21.000000,2.000000,15.000000,...,143.180000,183.550000,0.000000,70.000000,0.000000,0.000000,130.000000,19.760000,205.140000,382.000000
75%,8.000000,3.000000,2023-12-05 00:00:00,16044.000000,2023-12-11 00:00:00,16050.000000,1.000000,40.000000,2.000000,16.000000,...,405.040000,320.000000,210.000000,165.050000,111.680000,0.000000,259.110000,128.210000,298.630000,696.470000
max,18.000000,6.000000,2025-09-27 00:00:00,16706.000000,2025-09-29 00:00:00,16708.000000,79.000000,61.000000,12.000000,99.000000,...,17126.430000,32951.250000,16882.650000,8746.740000,4555.240000,1229.900000,18000.000000,8350.000000,3984.970000,9270.200000
std,4.410748,1.451564,NaN,1073.333074,NaN,1073.098260,3.610240,16.026343,2.020895,9.895404,...,612.523673,417.461472,897.014542,247.489248,171.487530,36.243747,484.252546,254.287360,191.494355,575.057786


In [27]:
df_emisivo.describe(include=object) # Resumen de las columnas de tipo object, que generalmente contienen texto

,Lugar Salida,Transporte Internacional de Salida,Pais,Departamento,Motivo,Ocupacion,Estudio,Destino,Alojamiento,Lugar Ingreso,Transporte Internacional de Ingreso,Trasporte Local
count,32109,32109,32109,32109,32109,32109,24406,32109,32109,32109,32109,32109
unique,16,6,21,20,11,25,8,13,15,14,6,7
top,Aeropuerto de Carrasco,Terrestre Auto,Uruguay,Montevideo,"Ocio, Recreo, Vacaciones","Empl. Adm, Cajero, Vendedor",Secundaria completa,Argentina,Hotel,Aeropuerto de Carrasco,Terrestre Auto,Auto propio
freq,6991,13967,31220,13513,15726,8269,8495,20910,14754,6994,13791,12658


##5. Verificación de valores faltantes

Identifica si hay columnas con valores faltantes y contabiliza cuántos hay en cada una. Proporciona una estrategia para tratar estos valores (puedes rellenarlos, eliminarlos o proponer una solución específica).

In [28]:
valores_nulos = df_emisivo.isna().sum() # Detectamos los valores nulos y contamos cuántos hay en cada columna

valores_nulos[valores_nulos > 0] # Filtramos para mostrar únicamente las columnas que tienen valores nulos

,0
IdNivelEstudio,7703
Estudio,7703


Los valores nulos se encuentran únicamente en **`IdNivelEstudio`** y **`Estudio`**, que representan la misma información mediante un código y su descripción. Se propone mantener los valores nulos en **`IdNivelEstudio`** y reemplazar los de **`Estudio`** por la categoría "Sin dato", ya que esto permite conservar todos los registros y facilita el análisis y la visualización de los datos sin alterar el significado de la variable.

In [29]:
df_emisivo["Estudio"] = df_emisivo["Estudio"].fillna("Sin dato") # Reemplazamos los valores nulos de la columna Estudio por "Sin dato"

df_emisivo["Estudio"].isna().sum() # Verificamos que funcionó correctamente

np.int64(0)

##6. Duplicados

Verifica si existen filas duplicadas en tu dataset. Justifica si decides eliminarlas o mantenerlas y realiza la acción correspondiente.

In [30]:
filas_duplicadas = df_emisivo.duplicated().sum() # Contamos la cantidad de filas duplicadas del DataFrame

print(f"El DataFrame tiene {filas_duplicadas} filas duplicadas.")

df_emisivo[df_emisivo.duplicated(keep=False)] # Mostramos las filas duplicadas

El DataFrame tiene 153 filas duplicadas.


,IdLugarSalida,Lugar Salida,IdTranspSalidad,Transporte Internacional de Salida,FechaSalida,IdFecSalida,FechaEntrada,IdFecEntrada,IdNacionalidad,Pais,...,GastoAlojamiento,GastoAlimentacion,GastoTransporteInternac,GatoTransporteLocal,GastoCultural,GastoTours,GastoCompras,GastoResto,Coef,CoefTot
430,10,Salto,2,Terrestre Auto,2017-01-05,13519,2017-01-10,13524,1,Uruguay,...,0.00,73.24,0.0,29.30,0.0,0.0,34.57,462.89,96.18,192.35
431,10,Salto,2,Terrestre Auto,2017-01-05,13519,2017-01-10,13524,1,Uruguay,...,0.00,73.24,0.0,29.30,0.0,0.0,34.57,462.89,96.18,192.35
490,7,Fray Bentos,3,Terrestre Bus,2017-01-12,13526,2017-01-17,13531,1,Uruguay,...,340.00,250.00,22.0,100.00,0.0,0.0,118.00,1580.00,128.81,257.63
491,7,Fray Bentos,3,Terrestre Bus,2017-01-12,13526,2017-01-17,13531,1,Uruguay,...,340.00,250.00,22.0,100.00,0.0,0.0,118.00,1580.00,128.81,257.63
499,3,Aeropuerto de Carrasco,1,Aereo,2017-01-06,13520,2017-01-12,13526,1,Uruguay,...,408.00,300.00,104.0,120.00,0.0,0.0,118.00,1896.00,129.04,258.09
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
31836,8,Paysandú,2,Terrestre Auto,2025-09-15,16694,2025-09-16,16695,1,Uruguay,...,0.00,0.00,0.0,0.00,0.0,0.0,0.00,0.00,460.75,460.75
31837,8,Paysandú,2,Terrestre Auto,2025-09-15,16694,2025-09-16,16695,1,Uruguay,...,0.00,0.00,0.0,0.00,0.0,0.0,0.00,0.00,460.75,460.75
31838,8,Paysandú,2,Terrestre Auto,2025-09-15,16694,2025-09-16,16695,1,Uruguay,...,0.00,0.00,0.0,0.00,0.0,0.0,0.00,0.00,460.75,460.75
31866,2,Puerto de montevideo,5,Maritimo - Fluvial,2025-09-17,16696,2025-09-19,16698,1,Uruguay,...,58.54,50.78,0.0,50.05,0.0,0.0,0.00,0.00,302.21,302.21


Se detectaron 153 filas duplicadas. Sin embargo, no es posible asegurar que esos registros correspondan a duplicados erróneos: cada fila representa a una persona y el dataset no cuenta con un identificador único que permita distinguir a los individuos. Eso significa que personas distintas pueden compartir exactamente los mismos valores en todas las variables registradas. Por eso se decidió mantener las filas duplicadas.

## 7. Valores únicos y su cantidad

Analiza las columnas categóricas para identificar qué valores únicos tienen y la moda. También, verifica cuántos valores únicos tiene cada columna del dataset.

Al observar este dataset, todas las columnas de tipo object corresponden a variables categóricas. Además, existen columnas numéricas que representan esas mismas categorías mediante un código identificador `(Id...)`, por lo que ambas contienen información redundante en distintos formatos. Para analizar los valores únicos y la moda, resulta más útil utilizar las columnas descriptivas de tipo object.

In [31]:
columnas_categoricas = df_emisivo.select_dtypes(include="object").columns # Seleccionamos las columnas de tipo object

for columna in columnas_categoricas: # Recorremos cada columna categórica con un bucle for
    print(f"\n{'='*50}") # \n genera un salto de línea y "=" * 50 crea un separador para mejorar la visualización
    print(f"Columna: {columna}")

    valores_unicos = df_emisivo[columna].unique() # Obtenemos los valores únicos de la columna
    print("\nValores únicos:")
    print(valores_unicos)

    conteo_unicos = df_emisivo[columna].nunique() # Contamos la cantidad de valores únicos
    print(f"\nCantidad de valores únicos: {conteo_unicos}")

    moda = df_emisivo[columna].mode()[0] # Obtenemos la moda de la columna
    print(f"\nModa: {moda}")


Columna: Lugar Salida

Valores únicos:
['Chuy' 'Río Branco' 'Aeropuerto de Carrasco' 'Fray Bentos' 'Bella Unión'
 'Colonia' 'Salto' 'Paysandú' 'Carmelo' 'Melo - Aceguá' 'Artigas' 'Rivera'
 'Puerto de montevideo' 'Nueva Palmira' 'Otros'
 'Aeropuerto de Punta del Este']

Cantidad de valores únicos: 16

Moda: Aeropuerto de Carrasco

Columna: Transporte Internacional de Salida

Valores únicos:
['Terrestre Auto' 'Aereo' 'Terrestre Bus' 'Maritimo - Fluvial'
 'Terrestre Otros' 'Otros']

Cantidad de valores únicos: 6

Moda: Terrestre Auto

Columna: Pais

Valores únicos:
['Uruguay' 'Brasil' 'Argentina' 'Otro de Europa' 'España' 'EE.UU.'
 'Italia' 'Paraguay' 'Chile' 'Ecuador' 'Venezuela' 'Bolivia'
 'Otro de America' 'Peru' 'Gran Bretaña' 'Colombia' 'Mexico' 'Francia'
 'Otro pais de Asia' 'Cuba' 'Africa u Oceania']

Cantidad de valores únicos: 21

Moda: Uruguay

Columna: Departamento

Valores únicos:
['Montevideo' 'Maldonado' 'Canelones' 'Salto' 'Lavalleja' 'Saqn Jose'
 'Rio Negro' 'Durazno' 'Ce

In [32]:
unicos_por_columna = df_emisivo.nunique().sort_values() # Verificamos la cantidad de valores únicos de todas las columnas del DataFrame y los ordenamos de manera ascendente
print(unicos_por_columna)

Transporte Internacional de Salida         6
IdTranspSalidad                            6
IdTranspIngreso                            6
Transporte Internacional de Ingreso        6
IdTranspLocal                              7
Trasporte Local                            7
IdNivelEstudio                             8
Estudio                                    9
IdMotivo                                  11
Motivo                                    11
Destino                                   13
IdDestino                                 13
Lugar Ingreso                             14
IdLugarIngreso                            14
Alojamiento                               15
IdAlojamiento                             15
Gente                                     15
Lugar Salida                              16
IdLugarSalida                             16
IdDeptoResidencia                         20
Departamento                              20
IdNacionalidad                            21
Pais      

## 8. Frecuencia de valores en columnas categóricas

Explora la frecuencia de los valores en al menos dos columnas categóricas importantes para comprender su distribución.

In [33]:
print('Frecuencia de valores en "Destino":') # Obtenemos la frecuencia de cada categoría de la columna "Destino"
print(df_emisivo["Destino"].value_counts())

print('\nFrecuencia de valores en "Motivo":') # Obtenemos la frecuencia de cada categoría de la columna "Motivo"
print(df_emisivo["Motivo"].value_counts())

print('\nFrecuencia de valores en "Pais":') # Obtenemos la frecuencia de cada categoría de la columna "Pais"
print(df_emisivo["Destino"].value_counts())

Frecuencia de valores en "Destino":
Destino
Argentina                   20910
Brasil                       6704
Centro y Norte America       1353
Europa                       1159
Paraguay                      605
Chile                         593
Resto Sud America             582
Asia del Este y Pacifico      108
Africa                         40
Oriente Medio                  29
Sin Datos                      21
Otros                           3
Asia Meridional                 2
Name: count, dtype: int64

Frecuencia de valores en "Motivo":
Motivo
Ocio, Recreo, Vacaciones      15726
Visita familiares / amigos    10440
Trabajo / Profesion            3507
Compras                        1182
Tratamiento Salud               308
Estudios                        276
Otros                           272
Religion y Peregrinaciones      184
Deportivo                       140
MICE                             55
Trabajo remunerado Destino       19
Name: count, dtype: int64

Frecuencia de valores 

## 9. Renombrar columnas

Si encuentras columnas con nombres confusos, renómbralas por nombres más claros y adecuados. Explica por qué realizaste los cambios.

In [34]:
for columna in df_emisivo.columns: # Mostramos el nombre de todas las columnas con un bucle for para mejor visualización
    print(columna)

IdLugarSalida
Lugar Salida
IdTranspSalidad
Transporte Internacional de Salida
FechaSalida
IdFecSalida
FechaEntrada
IdFecEntrada
IdNacionalidad
Pais
IdDeptoResidencia
Departamento
IdMotivo
Motivo
IdOcupacion
Ocupacion
IdNivelEstudio
Estudio
IdDestino
Destino
IdAlojamiento
Alojamiento
IdLugarIngreso
Lugar Ingreso
IdTranspIngreso
Transporte Internacional de Ingreso
IdTranspLocal
Trasporte Local
Estadia
Gente
GastoTotal
GastoAlojamiento
GastoAlimentacion
GastoTransporteInternac
GatoTransporteLocal
GastoCultural
GastoTours
GastoCompras
GastoResto
Coef
CoefTot


Se renombrarán algunas columnas para mejorar la claridad, corregir errores ortográficos y eliminar ambigüedades en los nombres. Los cambios a realizar son los siguientes:

*   **`Pais` a `PaisNacionalidad`**: La columna `Pais` en realidad representa la nacionalidad del viajero, no el país de destino, como se observó en la sección de frecuencia de valores. Se renombra para reflejar correctamente su contenido.
*   **`IdTranspSalidad` a `IdTranspSalida`**: Se corrige un error tipográfico.
*   **`GatoTransporteLocal` a `GastoTransporteLocal`**: Se corrige un error tipográfico.
*   **`TrasporteLocal` a `TransporteLocal`**: Se corrige un error tipográfico.
*   **`Coef` a `GastoCoefExpans`**: El nombre `Coef` es ambiguo. De acuerdo con los metadatos, esta columna representa el 'Coeficiente de expansión del gasto', por lo que se renombra para ser más explícito.
*   **`CoefTot` a `PersonaCoefExpans`**: Similar al anterior, `CoefTot` también es ambiguo. Los metadatos indican que representa el 'Coeficiente de expansión de las personas', por lo que se renombra para mayor claridad.

In [35]:
# Renombramos las columnas utilizando un diccionario, donde cada clave representa
# el nombre actual de la columna y cada valor el nuevo nombre que se le asignará

df_emisivo = df_emisivo.rename(columns={
    "Pais": "PaisNacionalidad",
    "IdTranspSalidad": "IdTransporteSalida",
    "Trasporte Local": "Transporte Local",
    "GatoTransporteLocal": "GastoTransporteLocal",
    "Coef": "CoefExpansionGasto",
    "CoefTot": "CoefExpansionPersonas"
})

for columna in df_emisivo.columns: # # Verificamos que los nuevos nombres de las columnas se hayan aplicado correctamente
    print(columna)


IdLugarSalida
Lugar Salida
IdTransporteSalida
Transporte Internacional de Salida
FechaSalida
IdFecSalida
FechaEntrada
IdFecEntrada
IdNacionalidad
PaisNacionalidad
IdDeptoResidencia
Departamento
IdMotivo
Motivo
IdOcupacion
Ocupacion
IdNivelEstudio
Estudio
IdDestino
Destino
IdAlojamiento
Alojamiento
IdLugarIngreso
Lugar Ingreso
IdTranspIngreso
Transporte Internacional de Ingreso
IdTranspLocal
Transporte Local
Estadia
Gente
GastoTotal
GastoAlojamiento
GastoAlimentacion
GastoTransporteInternac
GastoTransporteLocal
GastoCultural
GastoTours
GastoCompras
GastoResto
CoefExpansionGasto
CoefExpansionPersonas


##10. Ordenar el dataset

Ordena los datos según una o más columnas de tu elección (numéricas o categóricas). Explica por qué realizaste esta ordenación.

Se ordenará el DataFrame utilizando dos columnas:

*   **`FechaSalida` (ascendente)**: Permite visualizar los registros en orden cronológico, facilitando el análisis de la evolución temporal de los viajes.
*   **`GastoTotal` (descendente)**: Dentro de cada fecha de salida, los viajes con el mayor gasto aparecerán primero, lo que facilita identificar los registros de mayor valor y comparar el comportamiento del gasto entre los viajes realizados en una misma fecha.

In [36]:
# Ordenamos el DataFrame por la fecha de salida en forma ascendente y,
# dentro de cada fecha, por el gasto total en forma descendente

df_emisivo_ordenado = df_emisivo.sort_values(
    by=["FechaSalida", "GastoTotal"],
    ascending=[True, False]
    )

df_emisivo_ordenado.head() # Mostramos las primeras filas del DataFrame ordenado

,IdLugarSalida,Lugar Salida,IdTransporteSalida,Transporte Internacional de Salida,FechaSalida,IdFecSalida,FechaEntrada,IdFecEntrada,IdNacionalidad,PaisNacionalidad,...,GastoAlojamiento,GastoAlimentacion,GastoTransporteInternac,GastoTransporteLocal,GastoCultural,GastoTours,GastoCompras,GastoResto,CoefExpansionGasto,CoefExpansionPersonas
369,3,Aeropuerto de Carrasco,1,Aereo,2016-12-13,13496,2017-01-17,13531,1,Uruguay,...,0.0,165.09,0.0,0.00,52.41,0.0,1488.90,207.90,129.04,129.04
497,3,Aeropuerto de Carrasco,1,Aereo,2016-12-18,13501,2017-01-12,13526,1,Uruguay,...,0.0,1350.00,214.0,1000.00,0.00,0.0,2340.00,1700.00,129.04,258.09
9,4,Chuy,3,Terrestre Bus,2016-12-18,13501,2017-01-07,13521,1,Uruguay,...,0.0,514.47,0.0,225.08,96.46,0.0,67.52,96.46,89.61,179.22
360,1,Colonia,5,Maritimo - Fluvial,2016-12-19,13502,2017-01-04,13518,1,Uruguay,...,0.0,300.00,27.0,0.00,3.80,0.0,100.00,196.20,196.13,196.13
385,3,Aeropuerto de Carrasco,1,Aereo,2016-12-20,13503,2017-01-17,13531,1,Uruguay,...,0.0,419.16,0.0,0.00,329.34,0.0,2000.00,251.50,129.04,129.04


##11. Filtrado de datos

Crea al menos un filtro lógico para trabajar con un subconjunto de datos basado en condiciones específicas que consideres relevantes.

Se creará un subconjunto de datos para los viajes con las siguientes características:

*   **`Destino`: "Argentina"**: Argentina es el destino más frecuente para los viajeros uruguayos, por lo que analizar este segmento puede revelar patrones específicos de gasto y comportamiento.
*   **`Motivo`: "Ocio, Recreo, Vacaciones"**: Este es el motivo de viaje más común. Al combinarlo con el destino Argentina, podemos enfocar el análisis en los viajes de placer a este país, que representan una parte significativa del turismo emisivo.

In [37]:
# Filtramos los viajes con destino Argentina cuyo motivo es ocio, recreo o vacaciones
df_argentina_ocio = df_emisivo_ordenado[
    (df_emisivo_ordenado["Destino"] == "Argentina") &
    (df_emisivo_ordenado["Motivo"] == "Ocio, Recreo, Vacaciones")
]

print(f"Dimensiones del DataFrame filtrado: {df_argentina_ocio.shape}") # Mostramos las dimensiones del DataFrame filtrado

df_argentina_ocio.head() # Mostramos las primeras filas del DataFrame filtrado

Dimensiones del DataFrame filtrado: (9107, 41)


,IdLugarSalida,Lugar Salida,IdTransporteSalida,Transporte Internacional de Salida,FechaSalida,IdFecSalida,FechaEntrada,IdFecEntrada,IdNacionalidad,PaisNacionalidad,...,GastoAlojamiento,GastoAlimentacion,GastoTransporteInternac,GastoTransporteLocal,GastoCultural,GastoTours,GastoCompras,GastoResto,CoefExpansionGasto,CoefExpansionPersonas
444,1,Colonia,5,Maritimo - Fluvial,2016-12-21,13504,2017-01-02,13516,1,Uruguay,...,0.0,91.27,93.58,0.0,14.91,0.0,17.95,576.86,196.13,196.13
413,7,Fray Bentos,3,Terrestre Bus,2016-12-24,13507,2017-01-03,13517,1,Uruguay,...,0.0,350.50,189.27,0.0,10.54,0.0,0.00,339.95,128.81,257.63
483,1,Colonia,5,Maritimo - Fluvial,2016-12-25,13508,2017-01-03,13517,1,Uruguay,...,306.0,225.00,27.00,90.0,0.00,0.0,59.00,1422.00,196.13,196.13
515,1,Colonia,5,Maritimo - Fluvial,2016-12-26,13509,2017-01-03,13517,1,Uruguay,...,544.0,400.00,54.00,160.0,0.00,0.0,118.00,2528.00,196.13,392.26
512,10,Salto,2,Terrestre Auto,2016-12-26,13509,2017-01-02,13516,1,Uruguay,...,500.0,300.00,0.00,140.0,98.00,0.0,118.00,2212.00,96.18,192.35
